src/pkg/projects/Microsoft.NETCore.App/Microsoft.NETCore.App.pkgproj

```diff
@@ -80,6 +80,42 @@
                               PreferredPackages="$(Id);@(RuntimeDependency)" />
   </Target>
 
+  <UsingTask TaskName="GetVersionFromAssembly" AssemblyFile="$(LocalBuildToolsTaskDir)Microsoft.DotNet.Build.Tasks.Local.dll"/>
+
+  <Target Name="GenerateHashVersionsFile" BeforeTargets="GetPackageFiles">
+    <PropertyGroup>
+      <_SystemRuntimePath Condition="$([System.String]::new('%(File.Identity)').ToLowerInvariant().Contains('system.runtime.dll'))">%(File.Identity)</_SystemRuntimePath>
+      <_SystemPrivateCoreLibPath Condition="$([System.String]::new('%(File.Identity)').ToLowerInvariant().Contains('system.private.corelib.dll'))">%(File.Identity)</_SystemPrivateCoreLibPath>
+      <_SkipCoreClrHash Condition="'$(_SystemPrivateCoreLibPath)' == ''">true</_SkipCoreClrHash>
```

Comment:
You could avoid a lot of duplication by doing this in the depproj where we already identify all of these files: https://github.com/dotnet/core-setup/blob/f0fed403e0f455ebaf37626a9561b7de5fdbb160/src/pkg/projects/Microsoft.NETCore.App/src/Microsoft.NETCore.App.depproj#L30